File: TextAnalysisNetServer/Logics/WordCondition.cs
using System.Linq;
using System.Text.RegularExpressions;

namespace TextAnalysis
{
	public static class WordCondition
	{
		private static string prefixesPattern = @"^(re|dis|over|un|mis|out|de|fore|inter|pre|sub|trans|under)";
		private static string suffixesPattern = @"(ing|ed|ies)$";
		private static string rgx_acceptable_endings = @"(see|add|ass|ess|all|ill|ell|oll|iss|ree|ess|uzz|oss|lee|iff)$";
		private static string exceptionsPattern = @"(go|be|do)";
		private static string exceptionsEndingPattern = "(thing)$";
		private static string notContainsPattern = @"(with|the|and|she)";

		private static Regex prefixesRegex = new Regex(prefixesPattern);
		private static Regex suffixesRegex = new Regex(suffixesPattern);
		private static Regex exceptionsEndingRegex = new Regex(exceptionsEndingPattern);

		public static string GetClearWord(string word)
		{
			string tempWord;

			tempWord = prefixesRegex.Replace(word, "");

			if (!tempWord.Equals(word) && WordAcccepted(tempWord))
			{
				word = tempWord;
			}

			tempWord = suffixesRegex.Replace(word, "");

			if (!tempWord.Equals(word) && !Regex.IsMatch(word, exceptionsEndingPattern) && WordAcccepted(tempWord))
			{
				word = tempWord;
				if (word.Length > 2)
				{
					string ending = word.Substring(word.Length - 2);
					if (ending.Distinct().Count() == 1 && !Regex.IsMatch(word, rgx_acceptable_endings))
					{
						word = word.Substring(0, word.Length - 1);
					}
				}
			}
			return word;
		}

		public static bool WordAcccepted(string word){
			if(word!=null && !word.Equals("") && (word.Length>2 || Regex.IsMatch(word, exceptionsPattern))){
			  return true;
			}
			return false;
		}

		public static bool WordContainedNotIgnor(string word)
		{
			if (word != null && !word.Equals("") && !Regex.IsMatch(word, notContainsPattern))
			{
				return true;
			}
			return false;
		}
	}
}

File: TextAnalysisAndroid/app/src/main/java/com/likhtman/TextAnalysis/Logics/WordCondition.java
package com.likhtman.TextAnalysis.Logics;

import android.util.Log;

import java.util.regex.Pattern;

public class WordCondition {
    final String TAG = WordCondition.this.getClass().getSimpleName();

    private static WordCondition instance = null;

    private WordCondition(){

    }


    public static WordCondition Get_instance(){
        if(instance == null){
            instance = new WordCondition();
        }
        return instance;
    }


    private static String prefixesPattern = "^(re|dis|over|un|mis|out|de|fore|inter|pre|sub|trans|under)";
    private static String suffixesPattern = "(ing|ed|ies)$";
    private static String rgx_acceptable_endings = "(see|add|ass|ess|all|ill|ell|oll|iss|ree|ess|uzz|oss|lee|iff)$";
    private static String exceptionsPattern = "(go|be|do)";
    private static String exceptionsEndingPattern = "(thing)$";


    public String GetClearWord(String word)
    {
        word = word.replaceAll("#", "");


        String temp_word;

        temp_word = word.replaceAll(prefixesPattern, "");

        if(!temp_word.equals(word) && WordAcccepted(temp_word)){
            word = temp_word;
        }

        temp_word = word.replaceAll(suffixesPattern, "");

        if(!temp_word.equals(word) && !Pattern.compile(exceptionsEndingPattern).matcher(word).find() && WordAcccepted(temp_word)){
            word = temp_word;
            if (word.length() > 2)
            {
                String ending = word.substring(word.length() - 2);
                if (ending.chars().distinct().count() == 1 && !Pattern.compile(rgx_acceptable_endings).matcher(word).find())
                {
                    word = word.substring(0, word.length() - 1);
                }
            }
        }
        return word;
    }

    public boolean WordAcccepted(String word){
        if(word!=null && !word.equals("") && (word.length() > 2 || Pattern.compile(exceptionsPattern).matcher(word).find())){
            Log.d(TAG, "The word '" + word + "' accepted");
            return true;
        }
        Log.d(TAG, "The word '" + word + "' not accepted");
        return false;
    }

    

}
